istio-proxy/update.sh

```diff
@@ -4,68 +4,40 @@ set -o pipefail
 set -e
 set -u
 
-proxy_name=istio-proxy
-
 function usage() {
-	echo "Usage: $0 [-p <SHA of istio-proxy>]"
-	echo
-	exit 0
+    echo "Usage: $0 [-i <SHA of proxy>]"
+    echo
+    exit 0
 }
 
-while getopts ":p:" opt; do
-	case ${opt} in
-		p) PROXY_SHA="${OPTARG}";;
-		*) usage;;
-	esac
+while getopts ":i:" opt; do
+  case ${opt} in
+    i) SHA="${OPTARG}";;
+    *) usage;;
+  esac
 done
 
-PROXY_SHA=${PROXY_SHA:-"$(grep '%global proxy_git_commit ' ${proxy_name}.spec | cut -d' ' -f3)"}
+SHA=${SHA:-"$(grep '%global git_commit ' istio-proxy.spec | cut -d' ' -f3)"}
```

Comment:
Are we calling it istio-proxy.spec in product as well? The two files should match as much as possible.

Reply:
no

Reply:
that's the only difference - and the GN stuff now. But this GN should go away from product soon... well, that's my hope :)

Reply:
Then can we keep the variable please?

Reply:
It looks like GN is a metadata tool for Ninja: https://gn.googlesource.com/gn

Reply:
Done. I've added the variable back. Switched to upper case to denote a global variable instead of a local one. With the refactoring, there are only two occurrences of it now.

Reply:
Thanks. I'm approving.